### PR TITLE
Ensure archive_dir is present

### LINKED ISSUE #16: wildfly_install - failure when wildfly_archive_dir defined
Was able to successfully install several versions of Wildfly. Found that /opt/wildfly was getting a little cluttered up with .zip files so I thought I'd add a custom archive dir:

```
---
- name: "Wildfly installation and configuration"
  hosts:
    - buff
  become: yes
  vars:
    wildfly_version: '26.0.1.Final'
    wildfly_archive_dir: '{{ wildfly_install_workdir }}/archive'
  collections:
    - middleware_automation.wildfly
  roles:
    - wildfly_install
```

Defining `wildfly_archive_dir` caused the script to fail with:
```
TASK [middleware_automation.wildfly.wildfly_install : include_tasks] ***************************************
included: /Users/XXX/.ansible/collections/ansible_collections/middleware_automation/wildfly/roles/wildfly_install/tasks/install/wildfly.yml for buff

TASK [middleware_automation.wildfly.wildfly_install : assert] **********************************************
ok: [buff]

TASK [middleware_automation.wildfly.wildfly_install : Download Wildfly zipfile from {{ wildfly_install.download_url }} into {{ wildfly_install.path_to_archive }}] ***
ok: [buff -> localhost]

TASK [middleware_automation.wildfly.wildfly_install : include_tasks] ***************************************
skipping: [buff]

TASK [middleware_automation.wildfly.wildfly_install : include_tasks] ***************************************
skipping: [buff]

TASK [middleware_automation.wildfly.wildfly_install : Copy archive to target nodes] ************************
fatal: [buff]: FAILED! => {"changed": false, "checksum": "1f6676a7372256d5e2e80e5372b611a0e089a728", "msg": "Destination directory /opt/wildfly//archive does not exist"}
```

Have had a look at the script but don't understand it well enough to mess with it. Any suggestions?

### PR DESCRIPTION
**Pull request by aushacker**
Fixes #16 .

### CODE REVIEW

**Review by rpelisse — APPROVED** [review]

File: roles/wildfly_install/tasks/main.yml [hunk 1]
```diff
@@ -10,6 +10,13 @@
     owner: "{{ wildfly_install.user }}"
     group: "{{ wildfly_install.group }}"
 
+- name: "Ensures archive_dir {{ wildfly_install.archive_dir }} exists."
+  file:
+    path: "{{ wildfly_install.archive_dir }}"
+    state: directory
+    owner: "{{ wildfly_install.user }}"
+    group: "{{ wildfly_install.group }}"
+
```
Comment (guidograzioli) [inline roles/wildfly_install/tasks/main.yml:19]: It's missing the mode here, let's make it 0644

**Review by guidograzioli — CHANGES_REQUESTED** [review]

**Comment by rpelisse** [comment]
@aushacker can you udpate the PR? :)

**Comment by aushacker** [comment]
@rpelisse Rebased branch issue_16 to latest main and applied the requested change. Not sure what I've broken.

**Comment by guidograzioli** [comment]
> @rpelisse Rebased branch issue_16 to latest main and applied the requested change. Not sure what I've broken.

nothing serious broke, after you rebase you got the settings for the ansible-lint execution, and a linter is not happy about
a task name here https://github.com/ansible-middleware/wildfly/pull/19/files/fda0a25bb82edd4c33cd269f2694ac3ed5f2d412#diff-0046cb8a23c7e3b529a42af5ae1e2b70a389df9d466d320bae13811879e4218dR13 (wants "Ensure" not "Ensures; I know it may seem a silly rule, we should set it only as a warning)

**Comment by rpelisse** [comment]
@guidograzioli I think @aushacker refers to the fact that, now, this PR has **24 commits** in it. I think I will just cherry-pick the change and make a new PR.

**Comment by aushacker** [comment]
Thanks @rpelisse. I've messed up the rebase somehow and commits from the divergent main branch got included.

**Comment by rpelisse** [comment]
@aushacker don't worry, i'll cherry pick and make a new PR. Not sure when i'll come around it, but soon :)

**Comment by rpelisse** [comment]
Superseeded by [PR](https://github.com/ansible-middleware/wildfly/pull/25)

**Pull request closed without merging**